packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/connector/useConnector.tsx

```diff
@@ -41,17 +43,31 @@ export const useConnector = (): UseConnectorValue => {
   const theme = useTheme();
   const { hideDiagramElementPalette } = useDiagramElementPalette();
   const updateNodeInternals = useUpdateNodeInternals();
+  const { hoveredNode } = useContext<NodeContextValue>(NodeContext);
 
   const newConnectionStyleProvider: NodeStyleProvider = {
-    getNodeStyle: (id: string): React.CSSProperties => {
-      const nodeStyle: React.CSSProperties = {};
-      if (isNewConnection && candidates.map((node) => node.id).includes(id)) {
-        nodeStyle.boxShadow = `0px 0px 2px 2px ${theme.palette.primary.main}`;
-        nodeStyle.opacity = 1;
-      } else if (isNewConnection) {
-        nodeStyle.opacity = 0.4;
+    getNodeStyle: (id: string, descriptionId: string): React.CSSProperties => {
+      let opacity: string | undefined = undefined;
+      let boxShadow: string | undefined = undefined;
+      if (isNewConnection) {
+        const isConnectionCompatibleNode = Boolean(
+          candidates.find((nodeDescription) => nodeDescription.id === descriptionId)
+        );
+        const isSelectedNode = hoveredNode?.id === id;
+        if (isConnectionCompatibleNode) {
+          if (isSelectedNode) {
+            // Hightlight the selected target
+            boxShadow = `0px 0px 2px 2px ${theme.palette.primary.main}`;
+          }
+          // Make sure all compatible nodes, even normally faded ones, are fully visible
+          opacity = '1';
+        } else {
+          // Force fade all incompatible nodes
+          opacity = '0.4';
+        }
```

Comment:
Just curious since I don't see any issue with the result. But any reason why you did not keep the previous way of setting the values of the returned style ? It seems strange to me to set undefined values to filter them out later.

Reply:
You're right, the current version is halfway between the old version, which caused issues when combined with the style customizations from `useDropNode`, and what I think is a cleaner/easier to read *and* more reliable way.
I'll rework that.